File: frontend/src/services/orderService.js
import axios from "axios";

const API_URL = "http://localhost:8080/api/orders";

const getAuthHeaders = () => {
    const user = JSON.parse(localStorage.getItem("user"));
    return user?.token ? { Authorization: `Bearer ${user.token}` } : {};
};

const getAllOrders = async () => {
    const response = await axios.get(API_URL, { headers: getAuthHeaders() });
    return response.data;
};

const getOrderById = async (id) => {
    const response = await axios.get(`${API_URL}/${id}`, { headers: getAuthHeaders() });
    return response.data;
};

const getCustomerOrders = async (customerId) => {
    const response = await axios.get(`${API_URL}/customer/${customerId}`, { headers: getAuthHeaders() });
    return response.data;
};

const updateOrderStatus = async (id, status) => {
    const response = await axios.patch(
        `${API_URL}/${id}/status`,
        null,
        { params: { status }, headers: getAuthHeaders() }
    );
    return response.data;
};

const addTrackingDetails = async (id, trackingData) => {
    const response = await axios.put(
        `${API_URL}/${id}/tracking`,
        trackingData,
        { headers: getAuthHeaders() }
    );
    return response.data;
};

const cancelOrder = async (id) => {
    const response = await axios.post(
        `${API_URL}/${id}/cancel`,
        null,
        { headers: getAuthHeaders() }
    );
    return response.data;
};

const processPayment = async (id) => {
    const response = await axios.post(
        `${API_URL}/${id}/payment`,
        null,
        { headers: getAuthHeaders() }
    );
    return response.data;
};

const orderService = {
    getAllOrders,
    getOrderById,
    getCustomerOrders,
    updateOrderStatus,
    addTrackingDetails,
    cancelOrder,
    processPayment,
};

export default orderService;

File: ecommerce-backend/backend/src/main/java/com/ecommerce/service/OrderService.java
package com.ecommerce.service;

import com.ecommerce.dto.CheckoutDTO;
import com.ecommerce.dto.OrderDTO;
import com.ecommerce.dto.TrackingDTO;

import java.util.List;

public interface OrderService {
    OrderDTO checkout(CheckoutDTO checkoutDTO);
    OrderDTO getOrderById(Long orderId);
    List<OrderDTO> getCustomerOrders(Long customerId);
    List<OrderDTO> getAllOrders();
    OrderDTO updateOrderStatus(Long orderId, String status);
    OrderDTO addTrackingDetails(Long orderId, TrackingDTO trackingDTO);
    OrderDTO cancelOrder(Long orderId);
    OrderDTO processPayment(Long orderId);
}